src/Silverback.Integration/Messaging/Encryption/DecryptorConsumerBehavior.cs

```diff
@@ -38,9 +38,22 @@ public DecryptorConsumerBehavior(ISilverbackCryptoStreamFactory streamFactory)
             Check.NotNull(next, nameof(next));
 
             if (context.Envelope.Endpoint.Encryption != null &&
-                !context.Envelope.Headers.Contains(DefaultMessageHeaders.Decrypted) &&
+                !context.Envelope.Headers.Contains(DefaultMessageHeaders.Decrypted) && // TODO: Sergio when/where is it set?
```

Comment:
I guess you couldn't find where it's set and I'm honestly not sure this is still used. It was implemented like this to avoid re-decrypting a message when reprocessed via the `RetryPolicy` but now the mechanism is different and I don't think it's needed anymore.

Reply:
😄 well so I deleted the check.